src/extensions/renderer/canvas/drawing-nodes.js

```diff
@@ -342,6 +356,40 @@ CRp.drawNodeOverlay = function( context, node, pos, nodeWidth, nodeHeight ){
   }
 };
 
+CRp.drawNodeUnderlay = function( context, node, pos, nodeWidth, nodeHeight ){
```

Comment:
Could these underlay functions simply be made by making the overlay function generic?  E.g. a `prefix` parameter could be sent to the function so either `'overlay'` or `'underlay'` properties are used.  That way there's much less code